### DISCUSSION TITLE
creating and applying a library from GPF samples

### DISCUSSION THREAD

**Discussion by ekglover** [vdemichev/DiaNN#964] [Q&A]
Hi, I have used DiaNN with the library-free method on DIA samples but am now trying to analyse these same samples by first generating a library from a pooled sample run with gas-phase fractionation. I have noticed that when I input the GPF raw files and a FASTA file to generate the library and then later input my sample files and the library (no FASTA file loaded this time), the pg_matrix.tsv output has all empty fields for the first.protein.description column. Gene names etc are present. I do not need the first.protein.description data for my further analysis but does missing data suggest I have made a mistake? 

Secondly, I notice that even though FASTA digest, deep learning and the output option "Generate Spectral library" are all unticked in the second step (applying the GPF-generated library to my samples), in the DiaNN log the sample files are being used to supplement the provided library. Again, is this an error on my part? How is it doing this without a FASTA file being loaded - does it just call the FASTA file used for the GPF library prep? I can see that the GPF-generated library has been loaded and it recognises which FASTA file was used to create this. It seems that the GPF-library generated library is being combined with library-free method even though this is not selected and no FASTA file is loaded. 

Even though a spectral library is provided for protein quantification, the log reports one is also being created "A spectral library will be created from the DIA runs and used to reanalyse them; .quant files will only be saved to disk during the first step"

Many thanks for clarifying. 

Emily

**Comment by vdemichev** [comment]
Hi Emily,

If something is missing from the output, can you please share all the logs and I take a look? This usually happens because the FASTA database is not specified.

With GPF lib you'd want to disable MBR - I guess MBR would explain that sample files are used again to generate a new library.

You see this in the log because either MBR is enabled or DIA-NN is instructed to output a spectral library.

Best,
Vadim

**Reply by ekglover** [reply]
I see. I had MBR disabled when preparing the library from GPF files but enabled it to apply this library to quantify my samples. I can disable for quantification step as well. Should I specify the FASTA file even when quantifying with a prepared library rather doing library-free?

Logs attached. Many thanks for your explanations so far. 

Emily 

[20240301_HMIR21_PTEC_fGFPreport.log.txt](https://github.com/vdemichev/DiaNN/files/14589831/20240301_HMIR21_PTEC_fGFPreport.log.txt)
[20240301GFPlibreport.log.txt](https://github.com/vdemichev/DiaNN/files/14589841/20240301GFPlibreport.log.txt)

**Reply by vdemichev** [reply]
Hi Emily,

Yes, FASTA should always be specified if protein inference is enabled, which should be when using a DIA-based lib.
In the impending release this will be a bit different, but old DIA-NN versions benefit from FASTA.

Best,
Vadim

**Comment by ekglover** [comment]
Thank you. Those changes have solved things.
________________________________
